http_test.go

```diff
@@ -74,16 +77,24 @@ func (hc httpClientMock) Do(req *http.Request) (*http.Response, error) {
 
 func TestHTTPSourceClientSendUnit(t *testing.T) {
 	table := []struct {
-		category, hostname, sourcename *string
+		category, hostname, sourcename string
 	}{
-		{nil, nil, nil},
+		{"", "", ""},
 	}
 
 	for _, values := range table {
 		localUrl, _ := url.Parse(urlForUnitTests)
-		c := NewHTTPSourceClient(localUrl, values.category, values.hostname, values.sourcename)
+		c, err := NewHTTPSourceClient(localUrl,
+			SetXSumoCategoryHeader(values.category),
+			SetXSumoHostHeader(values.hostname),
+			SetXSumoNameHeader(values.sourcename),
+		)
+		if err != nil {
+			t.Fatalf("NewHTTPSourceClient got error: %s", err)
+		}
+
 		c.client = &httpClientMock{}
```

Comment:
I will add some behaviour to the mock so we can optionally request it to check for some headers values. Is that fine by you?

Reply:
Done in the next commit in this same branch ;)

Reply:
Thank you for adding many tests!

Reply:
You are welcome!